aim/tests/unit/test_aim_manager.py

```diff
@@ -48,7 +49,7 @@
 
 
 def getattr_canonical(obj, attr):
-    return base.sort_if_list(getattr(obj, attr))
+    return getattr(obj, attr)
```

Comment:
We should do "utils.deep_sort(getattr(obj, attr))" here I assume?

Reply:
`utils.deep_sort()` will be useful for comparing only when both the objects (which are being compared) are sorted by `utils.deep_sort()` because:

To maintain the items order we need to use `OrderedDict` and when we use it, it can be no longer directly compared to another regular `dict`.

Hence I created another boolean function `utils.is_equal()` to compare 2 objects using `utils.deep_sort()`.

I noticed that we are using `getattr_canonical()` only for comparing with some list of dict.

So I removed the `base.sort_if_list()` and instead used `utils.is_equal()` for comparing when we need to compare with `getattr_canonical()`